Xamarin.Forms.Platform.iOS/Renderers/NavigationRenderer.cs

```diff
@@ -319,6 +319,10 @@ protected virtual void OnElementChanged(VisualElementChangedEventArgs e)
 
 		protected virtual async Task<bool> OnPushAsync(Page page, bool animated)
 		{
+			if(page is MasterDetailPage)
+				System.Diagnostics.Debug.WriteLine($"Pushing a {nameof(MasterDetailPage)} onto a {nameof(NavigationPage)} is not a supported UI pattern on iOS. " +
```

Comment:
`Debug.WriteLine()` will be stripped out of release builds and users of the NuGet package won't see it.